### PR TITLE
Version V1.0.0

### PR DESCRIPTION
**Pull request by chuan-wang**
**Version 1.0.0**
(1) Support BCL Convert
(2) Remove redundant codes and obsoleted platforms
(3) Other refactors to improve performance or code readability

**Validation:**
23_30_TACA V1.0.0 with major revisions

**Known issue:**
BCL Convert does not support masks for more than 2 reads. Therefore cases that need three reads (e.g. 10X scATAC-seq) cannot be handled. In such cases the entire index 2 cycles will be ignored.

### CODE REVIEW

File: taca/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 """ Main TACA module
 """
 
-__version__ = "0.9.35"
\ No newline at end of file
+__version__ = "1.0.0"
```
Comment (alneberg) [inline taca/__init__.py:4]: Very nice!

File: taca/backup/backup.py [hunk 2]
```diff
@@ -377,10 +386,11 @@ def pdc_put(cls, run):
                     if bk._call_commands(cmd1='dsmc archive {}'.format(run.dst_key_encrypted), tmp_files=[run.flag]):
                         time.sleep(5) # give some time just in case 'dsmc' needs to settle
                         if bk.file_in_pdc(run.zip_encrypted) and bk.file_in_pdc(run.dst_key_encrypted):
-                            logger.info('Successfully sent file {} to PDC, removing file locally from {}'.format(run.zip_encrypted, run.path))
+                            logger.info('Successfully sent file {} to PDC, removing file locally from {} to archived folder'.format(run.zip_encrypted, run.path))
```
Comment (alneberg) [inline taca/backup/backup.py:389]:
```suggestion
                            logger.info('Successfully sent file {} to PDC, moving file locally from {} to archived folder'.format(run.zip_encrypted, run.path))
```

**Comment by alneberg** [comment]
Looking really good! Just a few thoughts:

1. Would it be possible to take this chance to split up `_aggregate_demux_results_simple_complex` into a few smaller chunks/functions to make it slightly easier to follow the logic?
2. Are the tests you remove actually outdated? We are looking into getting the tests running again...

File: taca/backup/backup.py (hunk 2 above)
Comment (chuan-wang) [reply taca/backup/backup.py]: Fxied

**Comment by chuan-wang** [comment]
> Looking really good! Just a few thoughts:
> 
> 1. Would it be possible to take this chance to split up `_aggregate_demux_results_simple_complex` into a few smaller chunks/functions to make it slightly easier to follow the logic?
> 2. Are the tests you remove actually outdated? We are looking into getting the tests running again...

Point 1 fixed. The removed tests are for the dismissed sequencing methods. We don't need them any more.

File: tests/test_illumina.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+NovaSeq#!/usr/bin/env python
```
Comment (alneberg) [inline tests/test_illumina.py:1]:
```suggestion
#!/usr/bin/env python
```

File: taca/illumina/Runs.py [hunk 4]
```diff
@@ -209,92 +238,16 @@ def _generate_per_lane_base_mask(self):
...
+    def _compute_base_mask(self):
         """
-            Assumptions:
-                - if runSetup is of size 3, then single index run
-                - if runSetup is of size 4, then dual index run
+        Assumptions:
+            - if runSetup is of size 3, then single index run
+            - if runSetup is of size 4, then dual index run
         """
         bm = []
```
[the remark this answers is not in the document]
Comment (alneberg) [reply taca/illumina/Runs.py:250]: I found the implementation in the standard run. It looks good. I would recommend to raise an not-implemented error for this one though.

**Review by alneberg — APPROVED** [review]
Amazing work! A few tiny comments/suggestions left, and then remember to resolve the conflicts as well.

**Comment by chuan-wang** [comment]
Validation 23_30 approved 2024-01-09. /CW

**Pull request merged**